src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs

```diff
@@ -1315,5 +1421,44 @@ public void RecordCloneSymbol()
             Assert.Equal(0, clone2.Arity);
             Assert.True(clone2.OverriddenMethod.Equals(clone, TypeCompareKind.ConsiderEverything));
         }
+
+        [Fact]
+        public void DataPropertiesInitOnly()
+        {
+            var src = @"
+class C
+{
+    data int X;
+    
+    public static void Main()
+    {
+        var c = new C() { X = 0 };
+        c.X = 2;
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (9,9): error CS8852: Init-only property or indexer 'C.X' can only be assigned in an object initializer, or on 'this' or 'base' in an instance constructor or an 'init' accessor.
+                //         c.X = 2;
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "c.X").WithArguments("C.X").WithLocation(9, 9));
+        }
+
+        [Fact]
+        public void DataPropertiesEmit()
+        {
+            var src = @"
+using System;
+class C
+{
+    data int X;
+    
+    public static void Main()
+    {
+        var c = new C() { X = 5 };
+        Console.WriteLine(c.X);
+    }
+}";
+            CompileAndVerify(src, expectedOutput: "5");
+        }
```

Comment:
I'm assuming we're not going to try and push this feature in for 16.7p3. We should take the time to flesh out more implementation and validation.

I'm seeing lots of places in prior code that special-case for `SyntaxKind.PropertyDeclaration`. They probably need to handle the new declaration kind.

Some test ideas:
- `override`, `sealed`, `abstract`, `extern` and other modifiers
- `data` property in generic type
-  `data` property initializer:
  - complex expression including `out` vars
  - `value` not in scope in initializer
  - verifying sequence points
- `GetDeclaredSymbol` on `data` property declaration, verify symbol 
- attributes on `data` property, including ones targeting underlying `[field: ...]`
- test at top-level
- parsing regular property or field returning type named "data`: `data field;`, `data Property => default;`
- verifying emitted metadata (PE symbols)